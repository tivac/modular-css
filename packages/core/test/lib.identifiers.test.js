"use strict";

var identifiers = require("../lib/identifiers.js");

describe("/lib", function() {
    describe("/identifiers.js", function() {
        it("should export a parse function", function() {
            expect(typeof identifiers.parse).toBe("function");
        });

        it("should export a keyframes regex", function() {
            expect(identifiers.keyframes).toBeInstanceOf(RegExp);
        });
        
        it("should extract a class", function() {
            expect(identifiers.parse(".fooga")).toMatchSnapshot();
        });
        
        it("should extract multiple classes", function() {
            expect(identifiers.parse(".fooga, .wooga")).toMatchSnapshot();
            expect(identifiers.parse(".fooga,.wooga")).toMatchSnapshot();
        });
        
        it("should extract an id", function() {
            expect(identifiers.parse("#fooga")).toMatchSnapshot();
        });
        
        it("should extract multiple ids", function() {
            expect(identifiers.parse("#fooga, #wooga")).toMatchSnapshot();
            expect(identifiers.parse("#fooga,#wooga")).toMatchSnapshot();
        });
        
        it("should extract @keyframes definitions", function() {
            expect(identifiers.parse("@keyframes fooga")).toMatchSnapshot();
            expect(identifiers.parse("@-webkit-keyframes fooga")).toMatchSnapshot();
            expect(identifiers.parse("@-moz-keyframes fooga")).toMatchSnapshot();
        });
        
        it("should extract multi-level classes", function() {
            expect(identifiers.parse(".fooga .wooga")).toMatchSnapshot();
        });
        
        it("should extract multi-level ids", function() {
            expect(identifiers.parse("#fooga #wooga")).toMatchSnapshot();
        });
        
        it("should extract mixed classes, ids, and keyframes", function() {
            // Slightly unintuitive output order is due to the iteration going classes then IDs
            // Also this selector is bonkers and makes no sense
            expect(
                identifiers.parse("#fooga, .wooga, #googa .booga, .tooga #looga, @keyframes kooga")
            )
            .toMatchSnapshot();
        });
    });
});
