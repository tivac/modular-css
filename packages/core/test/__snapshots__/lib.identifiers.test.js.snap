// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/lib /identifiers.js should extract @keyframes definitions 1`] = `
Array [
  "fooga",
]
`;

exports[`/lib /identifiers.js should extract @keyframes definitions 2`] = `
Array [
  "fooga",
]
`;

exports[`/lib /identifiers.js should extract @keyframes definitions 3`] = `
Array [
  "fooga",
]
`;

exports[`/lib /identifiers.js should extract a class 1`] = `
Array [
  "fooga",
]
`;

exports[`/lib /identifiers.js should extract an id 1`] = `
Array [
  "fooga",
]
`;

exports[`/lib /identifiers.js should extract mixed classes, ids, and keyframes 1`] = `
Array [
  "wooga",
  "booga",
  "tooga",
  "fooga",
  "googa",
  "looga",
  "kooga",
]
`;

exports[`/lib /identifiers.js should extract multi-level classes 1`] = `
Array [
  "fooga",
  "wooga",
]
`;

exports[`/lib /identifiers.js should extract multi-level ids 1`] = `
Array [
  "fooga",
  "wooga",
]
`;

exports[`/lib /identifiers.js should extract multiple classes 1`] = `
Array [
  "fooga",
  "wooga",
]
`;

exports[`/lib /identifiers.js should extract multiple classes 2`] = `
Array [
  "fooga",
  "wooga",
]
`;

exports[`/lib /identifiers.js should extract multiple ids 1`] = `
Array [
  "fooga",
  "wooga",
]
`;

exports[`/lib /identifiers.js should extract multiple ids 2`] = `
Array [
  "fooga",
  "wooga",
]
`;
